tests/test_checkout.py

```diff
@@ -8,64 +8,207 @@ class TestCheckout(object):
+        co = written_repo.checkout(write=True, branch_name=branch1)
+        co.datasets['_dset']['1'] = array5by7
+        co.metadata.add('a1', 'b1')
+        co.commit()
+        co.close()
+        branch2 = written_repo.create_branch('testbranch2')
+        co = written_repo.checkout(write=True, branch_name=branch2)
+        co.datasets['_dset']['2'] = array5by7
+        co.metadata.add('a2', 'b2')
+        co.commit()
+        co.close()
+        written_repo.merge('test merge 1', 'master', branch1)
+        written_repo.merge('test merge 2', 'master', branch2)
+        co = written_repo.checkout(branch_name='master')
+        assert len(co.datasets) == 1
+        assert len(co.datasets['_dset']) == 2
+        assert list(co.metadata.keys()) == ['a1', 'a2']
+
+    def test_merge_multiple_checkouts_multiple_dset(self, written_repo, array5by7):
+        branch1 = written_repo.create_branch('testbranch1')
+        co = written_repo.checkout(write=True, branch_name=branch1)
+        co.datasets['_dset']['1'] = array5by7
+        co.commit()
+        co.close()
+        branch2 = written_repo.create_branch('testbranch2')
+        co = written_repo.checkout(write=True, branch_name=branch2)
+        second_dset = co.datasets.init_dataset(name='second_dset', prototype=array5by7)
+        second_dset['1'] = array5by7
+        co.commit()
+        co.close()
+        written_repo.merge('test merge 1', 'master', branch1)
+        written_repo.merge('test merge 2', 'master', branch2)
+        co = written_repo.checkout(branch_name='master')
+        assert len(co.datasets) == 2
+        assert len(co.datasets['_dset']) == 1
+        assert len(co.datasets['second_dset']) == 1
+
+    def test_merge_diverged_conflict(self, written_repo, array5by7):
+        branch1 = written_repo.create_branch('testbranch1')
+        branch2 = written_repo.create_branch('testbranch2')
+        co = written_repo.checkout(write=True, branch_name=branch1)
+        co.datasets['_dset']['1'] = array5by7
+        co.metadata.add('a', 'b')
+        co.commit()
+        co.close()
+        co = written_repo.checkout(write=True, branch_name=branch2)
+        newarray = np.zeros_like(array5by7)
+        co.datasets['_dset']['1'] = newarray
+        co.metadata.add('a', 'c')
+        co.commit()
+        co.close()
+        written_repo.merge('commit message', 'master', branch1)
+        # TODO: This should raise a conflict coz of same key
+        written_repo.merge('commit message', 'master', branch2)
+        co = written_repo.checkout()
+        print(co.datasets['_dset']['1'])
+        print(co.metadata.get('a'))
+
+    def test_new_branch_from_where(self, written_repo, get_history, array5by7):
+        branch1 = written_repo.create_branch('testbranch1')
+        branch2 = written_repo.create_branch('testbranch2')
+        co1 = written_repo.checkout(write=True, branch_name=branch1)
+        # TODO: better way to get history: could be part of checkout itself
```

Comment:
agreed. please create issue and link it here.

Reply:
`repo.log(return_contents=True)` does this, right?

Reply:
yeah!